### DISCUSSION TITLE
Working of ExecuteScriptAsync() when used in a connected hostObject's method

### DISCUSSION THREAD

**Discussion by mounaj5** [MicrosoftEdge/WebView2Feedback#3974] [Q&A]
.js

```
 demo=async function f1(){
 var result= await window.chrome.webview.hostObjects.obj.SampleFunction(); 
 console.log(result); 
 return result; }
```

when demo() is called, result is null.
I spent some time looking into the code, looks like when SampleFunction() is invoked and as soon as
`await mWebView.ExecuteScriptAsync(script1)` is executed in .cs file, ` console.log(result)` is executed in .js without even waiting for SampleFunction() in .cs to complete and return the finalValue.
Besides, `await mWebView.ExecuteScriptAsync(script2)` doesn't give the expected value.
Please let me know if i'm doing something wrong.
I need to wait until SampleFunction() in .cs is complete,returns the finalValue and then execute `console.log(result)` in .js
Would be of great help if you could let me know how to do that. Thank you :)

.cs

```
      public async System.Threading.Tasks.Task<object> SampleFunction()
      {
         var value1 = await mWebView.ExecuteScriptAsync(script1);
         var value2 = await mWebView.ExecuteScriptAsync(script2);
        var finalValue = value1 + value2;  //any operations on value1 and value2
         return finalValue;
      }
```

**Comment by mounaj5** [comment]
Still waiting for an answer.
We are a little blocked by this. Can you please provide a reply ?

**Comment by david-risney** [comment]
Hi, sorry for the delay. I know we had added support for async .NET methods but then there were issues and we had to remove support. I'm not sure offhand if we've finished adding it back yet. Which version of the SDK and which version of the WebView2 Runtime are you using? Let me find someone who knows more about .NET async method support.

**Comment by mounaj5** [comment]
Hi,
This behavior was seen in WebView2 SDK 1.0.1248-prerelease version.
Moved to WebView2 SDK 1.0.1293.44 (runtime 104.0.1293.70), recently and noticed that ExecuteScriptAsync() inside connected host object's method is rendering the control unusable.
1. Is it not allowed to use ExecuteScriptAsync() (or re-enter js context) in a .NET method which is called from js ?

This was possible in webBrowser Control.
We have used mWebBrowser.Document.InvokeScript(script1, args);  in a .NET method which is called from js through window.external.
2. If webView2 doesn't provide DOM support and if we have to run ExecuteScriptAsync() for the same, how to we access DOM elements or invoke script inside a connected object's method  ?

Anyway, fortunately we had a workaround to achieve the same without having to call scripts inside of a connected object's method. Answers to the above issues would be helpful for us in the future.
Thank you :)
